src/ImageSharp.Web/TagHelpers/ImageTagHelper.cs

```diff
@@ -0,0 +1,378 @@
+            this.AddProcessingCommands(context, output, commands, this.parserCulture);
+
+            byte[] secret = this.options.HMACSecretKey;
+            if (commands.Count > 0 || secret?.Length > 0)
+            {
+                // Retrieve the TagHelperOutput variation of the "src" attribute in case other TagHelpers in the
+                // pipeline have touched the value. If the value is already encoded this helper may
+                // not function properly.
+                src = output.Attributes[SrcAttributeName].Value as string;
+                if (secret?.Length > 0)
+                {
+                    string hash = this.authorizationUtilities.ComputeHMAC(src, commands, secret);
+                    commands.Add(RequestAuthorizationUtilities.TokenCommand, hash);
+                }
+
+                src = AddQueryString(src, commands);
+            }
+
+            this.Src = src;
+            output.Attributes.SetAttribute(SrcAttributeName, src);
+        }
+
+        /// <summary>
+        /// Allows the addition of processing commands by inheriting classes.
+        /// </summary>
+        /// <param name="context">Contains information associated with the current HTML tag.</param>
+        /// <param name="output">A stateful HTML element used to generate an HTML tag.</param>
+        /// <param name="commands">The command collection.</param>
+        /// <param name="commandCulture">The culture to use when generating and processing commands.</param>
+        protected virtual void AddProcessingCommands(
+            TagHelperContext context,
+            TagHelperOutput output,
+            CommandCollection commands,
+            CultureInfo commandCulture)
+        {
+            this.AddResizeCommands(output, commands);
+            this.AddAutoOrientCommands(commands);
+            this.AddFormatCommands(commands);
+            this.AddBackgroundColorCommands(commands);
+            this.AddQualityCommands(commands);
+        }
+
+        private void AddResizeCommands(TagHelperOutput output, CommandCollection commands)
+        {
+            // If no explicit width/height has been set on the image, set the attributes to match the
+            // width/height from the process commands if present.
+            int? width = output.Attributes[ResizeWebProcessor.Width]?.Value as int?;
+            if (this.Width.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Width, this.Width.Value.ToString(this.parserCulture));
+                output.Attributes.SetAttribute(ResizeWebProcessor.Width, width ?? this.Width);
+            }
+
+            int? height = output.Attributes[ResizeWebProcessor.Height]?.Value as int?;
+            if (this.Height.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Height, this.Height.Value.ToString(this.parserCulture));
+                output.Attributes.SetAttribute(ResizeWebProcessor.Height, height ?? this.Height);
+            }
```

Comment:
The attribute value of the width/height will most likely be a `string` (not an `int?`), so it currently still overwrites the attribute values. There's also no reason to get this value if the `imagescharp-` prefixed values aren't set, so this can be rewritten as:

```suggestion
            if (this.Width.HasValue)
            {
                commands.Add(ResizeWebProcessor.Width, this.Width.Value.ToString(this.parserCulture));

                if (!output.Attributes.ContainsName(ResizeWebProcessor.Width))
                {
                    output.Attributes.SetAttribute(ResizeWebProcessor.Width, this.Width.Value);
                }
            }
            
            if (this.Height.HasValue)
            {
                commands.Add(ResizeWebProcessor.Height, this.Height.Value.ToString(this.parserCulture));

                if (!output.Attributes.ContainsName(ResizeWebProcessor.Height))
                {
                    output.Attributes.SetAttribute(ResizeWebProcessor.Height, this.Height.Value);
                }
            }
```

Reply:
>The attribute value of the width/height will most likely be a string (not an int?)

No, they're not. The specification described width and height of image elements as representative of CSS pixel units. They're not strings.

Reply:
We always want to read the values anyway and compare as an explicit with might have been set that is different to the requested width. (Think downscaling for high resolution)